Source/VMware.vSphereDSC/DSCResources/VMHostNetwork/VMHostVss/VMHostVssBridge.ps1

```diff
@@ -0,0 +1,197 @@
+#>
+
+[DscResource()]
+class VMHostVssBridge : VMHostVssBaseDSC {
+    <#
+    .DESCRIPTION
+
+    The list of keys of the physical network adapters to be bridged.
+    #>
+    [DscProperty()]
+    [string[]] $NicDevice
+
+    <#
+    .DESCRIPTION
+    The beacon configuration to probe for the validity of a link.
+    If this is set, beacon probing is configured and will be used.
+    f this is not set, beacon probing is disabled.
+    Determines how often, in seconds, a beacon should be sent.
+    #>
+    [DscProperty()]
+    [int] $BeaconInterval
+
+    <#
+    .DESCRIPTION
+
+    The link discovery protocol, whether to advertise or listen .
+    #>
+    [DscProperty()]
+    [LinkDiscoveryProtocolOperation] $LinkDiscoveryProtocolOperation
+
+    <#
+    .DESCRIPTION
+
+    The link discovery protocol type.
+    #>
+    [DscProperty()]
+    [LinkDiscoveryProtocolProtocol] $LinkDiscoveryProtocolProtocol
+
+    [void] Set() {
+        Write-Verbose -Message "$(Get-Date) $($s = Get-PSCallStack; "Entering {0}" -f $s[0].FunctionName)"
+
+        $this.ConnectVIServer()
+        $vmHost = $this.GetVMHost()
+        $this.GetNetworkSystem($vmHost)
+
+        $this.UpdateVssBridge($vmHost)
+    }
+
+    [bool] Test() {
+        Write-Verbose -Message "$(Get-Date) $($s = Get-PSCallStack; "Entering {0}" -f $s[0].FunctionName)"
+
+        $this.ConnectVIServer()
+        $vmHost = $this.GetVMHost()
+        $this.GetNetworkSystem($vmHost)
+        $vss = $this.GetVss()
+
+        if ($this.Ensure -eq [Ensure]::Present) {
+            return ($null -ne $vss -and $this.Equals($vss))
+        }
+        else {
+            $this.NicDevice = @()
+            $this.BeaconInterval = 0
+            $this.LinkDiscoveryProtocolProtocol = [LinkDiscoveryProtocolProtocol]::Unset
+
+            return ($null -eq $vss -or $this.Equals($vss))
+        }
+    }
+
+    [VMHostVssBridge] Get() {
+        Write-Verbose -Message "$(Get-Date) $($s = Get-PSCallStack; "Entering {0}" -f $s[0].FunctionName)"
+
+        $result = [VMHostVssBridge]::new()
+        $result.Server = $this.Server
+
+        $this.ConnectVIServer()
+        $vmHost = $this.GetVMHost()
+        $this.GetNetworkSystem($vmHost)
+
+        $result.Name = $vmHost.Name
+        $this.PopulateResult($vmHost, $result)
+
+        $result.Ensure = if ([string]::Empty -ne $result.VssName) { 'Present' } else { 'Absent' }
+
+        return $result
+    }
+
+    <#
+    .DESCRIPTION
+
+    Returns a boolean value indicating if the VMHostVssBridge should to be updated.
+    #>
+    [bool] Equals($vss) {
+        Write-Verbose -Message "$(Get-Date) $($s = Get-PSCallStack; "Entering {0}" -f $s[0].FunctionName)"
+
+        $typeName = 'VMware.Vim.HostVirtualSwitchBondBridge'
```

Comment:
Here you can simplify the whole logic if you define the following hidden property in the class:
```
hidden [string] $bridgeType = 'HostVirtualSwitchBondBridge'
```

And then use it in the Test method:
```
$vss.Spec.Bridge.GetType().Name -eq $this.bridgeType
```

Reply:
I don't really agree with this one.
You're abolishing the -is operator to verify the correct type, in favour of a string comparison.
I'm of the opinion that the -is operator is the correct way to test a Type.

Reply:
Yes I agree that -is operator is far better solution than the one with strings. One way we can accomplish that is if we add VMware.Vim module to the ImportRequiredModules() function to avoid the try/catch block here. And then separate the TestHelpers into separate modules: (VMware.Vim, VMware.VimAutomation.Core and so on). Currently I am working on the generation of the types, so when they are ready we can refactor the helper module, which would allow us to use the -is operator in the Resources.

Reply:
Updated